dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/MessageLogHandler.java

```diff
@@ -52,14 +51,8 @@ public void synchronousWrite(Object event) {
          * Given an 'object' we must determine what type of log event it originates from.
          * Knowing that it is a *Data object, we can figure what type of source it is.
          */
-        GenericData genData = null;
 
-        //check if event is a LogTraceData, if it is then we must extract the genericData.
-        if (event instanceof LogTraceData) {
-            genData = ((LogTraceData) event).getGenData();
-        } else if (event instanceof GenericData) {
-            genData = (GenericData) event;
-        }
+        GenericData genData = (GenericData) event;
```

Comment:
at start of this method, need to check to make sure event is instanceof GenericData.  If it's not GenericData we should throw an IllegalArgumentException

Reply:
ok